sky/skylet/events.py

```diff
@@ -211,7 +211,12 @@ def _stop_cluster_with_new_provisioner(self, autostop_config,
 
         # Stop the ray autoscaler to avoid scaling up, during
         # stopping/terminating of the cluster.
-        subprocess.run('ray stop', shell=True, check=True)
+        proc = subprocess.run('ray stop', shell=True, check=False)
+        if proc.returncode != 0:
+            operator_str = 'terminate' if autostop_config.down else 'stop'
+            logger.warning('Failed to stop ray autoscaler. Continuing to '
+                           f'{operator_str} the cluster. Please be careful of '
+                           'any possible leaked VMs.')
```

Comment:
This may be risky since it can be a silent leak. Versus a clearly visible autostop/down failure.

Reply:
It should be similar to autostop/down failure, as both will still have an entry in the local status table. By skipping the failure of `ray stop`, we have more chance to successfully stop/terminate the cluster. Wdyt?

Reply:
Not sure I follow. Let's assume we're auto-downing, and assume `ray stop` failed and the autoscaler hasn't been stopped. We then proceed to L229 to down all workers on the cloud. Autoscaler notices it and starts provisioning new workers. We then proceed to L232 to down the head node. At this point the cluster table's record will have been cleaned up while there are leaked VMs on the cloud?
 
It seems for auto-stopping, if the above events happen, the cluster table will have an INIT record, which is not as bad as the auto-down case.

Reply:
> At this point the cluster table's record will have been cleaned up while there are leaked VMs on the cloud?

The cluster table's record will not be automatically cleaned up, instead if the user or controller do `sky status -r` locally, skypilot will find out that there is still some nodes with the same cluster_name on the cloud and set the cluster to INIT.

That said, I agree that allowing `ray stop` to have non-zero return code may not worth it, as it seems `ray stop` will always return 0 for all the case I tested:
1. no ray cluster exists
2. ray cluster exists
3. user custom ray cluster exists and fail to kill due to permission error.